test/native/integration-test-helpers/rich-text-change-and-select-text.js

```diff
@@ -3,20 +3,42 @@
  */
 import { fireEvent } from '@testing-library/react-native';
 
+let eventCount = 0;
+
+function stripOuterHtmlTags( string ) {
+	return string.replace( /^<[^>]*>|<[^>]*>$/g, '' );
+}
+
+function insertTextAtPosition( text, newText, start, end ) {
+	return text.slice( 0, start ) + newText + text.slice( end );
+}
+
 /**
  * Changes the text and selection of a RichText component.
  *
- * @param {import('react-test-renderer').ReactTestInstance} richText                 RichText test instance.
- * @param {string}                                          text                     Text to be set.
- * @param {Object}                                          options                  Configuration options for selection.
- * @param {number}                                          [options.selectionStart] Selection start position.
- * @param {number}                                          [options.selectionEnd]   Selection end position.
+ * @param {import('react-test-renderer').ReactTestInstance} richText                        RichText test instance.
+ * @param {string}                                          text                            Text to be set.
+ * @param {Object}                                          options                         Configuration options for selection.
+ * @param {number}                                          [options.initialSelectionStart]
+ * @param {number}                                          [options.initialSelectionEnd]
+ * @param {number}                                          [options.selectionStart]        Selection start position.
+ * @param {number}                                          [options.selectionEnd]          Selection end position.
  */
 export const changeAndSelectTextOfRichText = (
 	richText,
 	text,
-	{ selectionStart = 0, selectionEnd = 0 } = {}
+	options = {}
 ) => {
+	const currentValueSansOuterHtmlTags = stripOuterHtmlTags(
+		richText.props.value
+	);
+	const {
+		initialSelectionStart = currentValueSansOuterHtmlTags.length,
+		initialSelectionEnd = initialSelectionStart,
+		selectionStart = 0,
+		selectionEnd = selectionStart,
```

Comment:
Yes, great idea! 

This somewhat relates to another thought that has crossed my mind: should we merge `changeTextOfRichText` and `changeAndSelectTextOfRichText`?

The nuanced difference between them is an implementation detail. Specifically, the fact that `onSelectionChange` invokes `onTextUpdate`. I think it could be simpler to remove the current `changeTextOfRichText` and rename `changeAndSelectTextOfRichText` to `changeTextOfRichText`. 

This would mean `changeTextOfRichText` would default to mimicking typing in an input — both in appending text and placing the cursor at the end of the typed value — while still retaining the ability to manually set initial or final selection positions. 

WDYT?

Reply:
I had the same thought, I don't think we really need two functions so we could unify them. I think I coupled those helpers too close to the event logic of `RichText` 😅. Ideally, I think we could have two helpers to manipulate that component:
- `typeTextInRichText`: Inserts the provided text, the inserting position will be based on the selection values of the text input. The function will update the selection values to point the inserting position always at the end of the inserted text.
- `selectTextInRichText`: Updates the selection values of the text input. This could be combined with the above function to replace parts of the content.

Reply:
Great, thank you very much 🙇 !